capstone/backend/src/main/java/com/google/spanner/CreateSpannerTables.java

```diff
@@ -0,0 +1,157 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.java.spanner;
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.spanner.Database;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.InstanceAdminClient;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.spanner.admin.database.v1.CreateDatabaseMetadata;
+import java.util.Arrays;
+import java.util.concurrent.ExecutionException;
+import org.threeten.bp.LocalDate;
+import org.threeten.bp.temporal.ChronoField;
+
+
+public class UsersSpanner {
+
+  /*
+   * Creates database using an Admin Client, accessing the instance signalled in
+   * the DatabaseId and creating 5 hard coded tables: Users, UserLists, Stores, Items,
+   * and Inventory.
+   * 
+   * @param dbAdminClient client used to access the instance and database
+   * @param id contains the information regarding the instance (name, etc.)
+   *
+   * @throws SpannerException if failed during execution
+   */
+  static void createDatabase(DatabaseAdminClient dbAdminClient, DatabaseId id) {
+    OperationFuture<Database, CreateDatabaseMetadata> op =
+        dbAdminClient.createDatabase(
+            id.getInstanceId().getInstance(),
+            id.getDatabase(),
+            Arrays.asList(
+                "CREATE TABLE Users ("
+                    + "  UserId    INT64,"
+                    + "  Username  STRING(MAX),"
+                    + "  Email     STRING(MAX)"
+                    + ") PRIMARY KEY (UserId)",
+                "CREATE TABLE UserLists ("
+                    + "  UserId      INT64,"
+                    + "  ListId      INT64,"
+                    + "  ItemTypes   ARRAY<STRING(MAX)>"
+                    + ") PRIMARY KEY (UserId, ListId),"
+                    + "  INTERLEAVE IN PARENT Users ON DELETE CASCADE",
+                "CREATE TABLE Stores ("
+                    + "  StoreId      INT64,"
+                    + "  StoreName    STRING(MAX),"
+                    + "  Address      STRING(MAX)"
+                    + ") PRIMARY KEY (StoreId)",
+                "CREATE TABLE Items ("
+                    + "  ItemId            INT64,"
+                    + "  ItemNameAndBrand  STRING(MAX),"
+                    + "  ItemType          STRING(MAX)"
+                    + ") PRIMARY KEY (ItemId)",
+                "CREATE TABLE Inventory ("
```

Comment:
I think a singular Inventory makes sense because while each store has its own inventory, this is a list of ALL the items from ALL the stores, so it's still one bigger inventory in a sense? But I'm not strongly opposed to "Inventories"; it just looks a little weird in my opinion.

Reply:
Yeah, either all plural or all singular is good. "Inventories" is a bit weird but it's nice to keep it consistent.